biogeochem/FatesLitterMod.F90

```diff
@@ -0,0 +1,429 @@
+module FatesLitterMod
+
+  ! -------------------------------------------------------------------------------------
+  ! This module contains methods and type definitions for all things litter.
+  ! "litter" means all organic material that is no longer associated with a live plant.
+  ! Also, in FATES we only track "un-fragmented" and "un-decomposed" litter. This
+  ! is a decision of pragmatism, as FATES is not a soil decomposition model, yet FATES
+  ! does need to retain litter for fire calculations.  THerefore, we raitain
+  ! undecomposed litter for a period of time in FATES, until it fragments and is passed
+  ! to another model to handle deocomposition.
+  ! 
+  ! This encompasses:  1) "Coarse Woody Debris"
+  !                    2) fine materials leaves, roots etc 
+  !                       (sometimes exclusively refered to as litter)
+  !                    3) Reproductive materials (seeds, nuts, fruits)
+  !
+  ! Important point:   THESE POOLS DO NOT CONTAIN DECOMPOSING MATTER !!!!!
+  !
+  ! Another Important Point: We track the fine litter by its "decomposability" pool.
+  !                          However, we don't actually apply any differential
+  !                          turnover rates based on these pools, we are just
+  !                          differentiating, tracking and preserving them to be 
+  !                          passed in the correct partitions to the BGC model. 
+  !                          Their partitions are a PFT parameter.
+  ! 
+  ! -------------------------------------------------------------------------------------
+
+
+   ! To-do:
+   ! 8) In CWD_IN, add the flux diagnostics, then remove the
+   !    patch level rate in the history code
+
+   
+   use FatesConstantsMod, only : r8 => fates_r8
+   use FatesConstantsMod, only : i4 => fates_int
+   use FatesConstantsMod, only : nearzero
+   use FatesConstantsMod, only : calloc_abs_error
+   use FatesConstantsMod, only : fates_unset_r8
+
+   use FatesGlobals     , only : endrun => fates_endrun
+   use FatesGlobals     , only : fates_log 
+   use shr_log_mod      , only : errMsg => shr_log_errMsg
+
+   implicit none
+   private
+
+
+   integer, public, parameter :: ncwd  = 4    ! number of coarse woody debris pools 
+                                              ! (twig,s branch,l branch, trunk)
+
+   integer, public, parameter :: ndcmpy = 3   ! number of "decomposability" pools in
+                                              ! fines (lignin, cellulose, labile)
```

Comment:
@jkshuman ,
All litter, including CWD will ultimately be partitioned into the 3 decomposability pools.  So this comment should be changed.  
I must had put "fines" in there probably because we are only tracking those 3 pools in FATES for the fine tissues (as prognostic variables), whereas for the CWD we partition into those pools only as it is being sent from FATES to the BGC model.